### PR TITLE
Rewrite of plugin

### LINKED ISSUE #52: Rewrite of the plugin
Hi @szibis 

I wanted to show you a rewrite I have done of helm-secrets.
https://github.com/mhyllander/helm-secrets/tree/fixes

I find the plugin idea very useful and it would have solved the problems I and my team had when starting to use kubernetes and helm. But I had some problems using it, so I started on a fork and ended up rewriting most of the plugin.

1. There is some confusion in the current plugin whether it should decrypt template files or not. Documentation and implementation are not in sync. My design decision is that template files should not contain secrets. Secrets should only be stored in value files, by convention named "secrets.yaml" (or "secrets.\*.yaml" if you need several). The decrypted files are named "secrets.dec.yaml" or "secrets.\*.dec.yaml".
2. I have included the wrapper inside the plugin, so instead of running "helm-wrapper install ...", you run "helm secrets install ...". Etc. The wrapper function does intelligent parsing of command line options.
3. The wrapper no longer scans the chart for files to decrypt. It only decrypts value files specified with the -f and --values options on the command line (and removes the on-the-fly decrypted files after the helm command has been run).

I realize my changes are probably too much to be merged back to this repo (and maybe I haven't encountered all use cases), but I wanted to notify you about my fork in case you were interested.

### PR DESCRIPTION
**Pull request by mhyllander**
as detailed in #52 

fixes #52

### CODE REVIEW

File: secrets.sh [hunk 1]
```diff
@@ -1,363 +1,496 @@
...
+	case "$1" in
+	    --)
+		# skip --, and what remains are the cmd args
+		shift 
+		break
+		;;
+            -f|--values)
+		cmdopts+=("$1")
+		yml="$2"
+		if [[ $yml =~ ^(.*/)?secrets(\.[^.]+)\.yaml$ ]]
```
Comment (pesimon) [inline secrets.sh]: can we make this
`if [[ $yml =~ ^(.*/)?secrets(\.[^.]+)*\.yaml$ ]]`
to match `secrets.yaml` and `secrets.bla.yaml`
 
It seams to be a common practise to have Config / GitOps repo like: 
`<env>/<chartname>/{values,secrets}.yaml`
which could support this opinionated structure too.

Comment (mhyllander) [reply secrets.sh]: Thanks for spotting that!

**Comment by monotek** [comment]
Any update on this?

**Comment by mhyllander** [comment]
@szibis I think this is pretty stable now. I use it daily and have fixed a number of issues since I originally created the PR.

**Comment by szibis** [comment]
Sorry for long silence from my side. I will try today/tomorrow review all and merge to master.

**Comment by szibis** [comment]
Sorry for long silence on my side. I will try today/tomorrow review all and merge to master.

**Comment by szibis** [comment]
Sorry for long silence on my side. I will try today/tomorrow review all and merge to master.

**Comment by szibis** [comment]
Sorry for long silence on my side. I will try today/tomorrow review all and merge to master.

**Comment by szibis** [comment]
Thx, based on this we will make changes in other project and some feature for backward compatibility before releasing new 2.0.0 version.

**Pull request merged**